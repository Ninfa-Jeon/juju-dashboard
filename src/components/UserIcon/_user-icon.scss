@import "vanilla-framework/scss/vanilla";
@import "../../scss/functions/z-index";

.user-icon {
  cursor: pointer;
  display: grid;
  margin-left: -0.75rem;
  position: relative;
  text-align: center;

  &__panel {
    @include vf-animation(opacity, brisk, ease-in-out);

    min-width: 100px;
    opacity: 0;
    overflow: visible;
    padding: 0.25rem;
    position: absolute;
    right: 0.5rem;
    top: 2.5rem;

    .p-list {
      margin-bottom: 0;
    }

    &.is-visible {
      opacity: 1;
      z-index: z("gamma");
    }

    &-arrow {
      background: $color-x-light;
      box-shadow: 3px 3px 6px -3px rgba(0, 0, 0, 0.2);
      height: 1rem;
      position: absolute;
      right: 0.75rem;
      top: -0.25rem;
      transform: rotate(225deg);
      width: 1rem;
      z-index: z("gamma");
    }
  }

  .p-icon--user {
    height: 1.5rem;
    margin: auto;
    width: 1.5rem;
  }
}
