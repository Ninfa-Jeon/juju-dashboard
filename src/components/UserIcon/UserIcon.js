import React, { useState } from "react";
import classNames from "classnames";
import { Link } from "react-router-dom";

import { logOut } from "app/actions";

import "./_user-icon.scss";

export default function User(props) {
  const [userPanelVisibility, setUserPanelVisibility] = useState(false);

  return (
    <div className="user-icon">
      <i
        className="p-icon--user"
        onClick={() => setUserPanelVisibility(!userPanelVisibility)}
      >
        Account icon
      </i>
      <div
        className={classNames("user-icon__panel p-card--highlighted", {
          "is-visible": userPanelVisibility
        })}
      >
        <span className="user-icon__panel-arrow"></span>
        <ul className="p-list">
          <li className="p-list__item">
            <a href="#_">Profile</a>
          </li>
          <li className="p-list__item">
            <a href="#_">Help</a>
          </li>
          <li className="p-list__item">
            <Link to="/" onClick={() => logOut()}>
              Log out
            </Link>
          </li>
        </ul>
      </div>
    </div>
  );
}
