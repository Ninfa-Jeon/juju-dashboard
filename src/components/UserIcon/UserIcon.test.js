import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import dataDump from "testing/complete-redux-store-dump";

import UserIcon from "./UserIcon";

const mockStore = configureStore([]);
describe("User Icon", () => {
  it("renders without crashing and matches snapshot", () => {
    const store = mockStore(dataDump);
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <UserIcon />
        </Router>
      </Provider>
    );
    expect(wrapper.find(".user-icon")).toMatchSnapshot();
  });

  it("toggles drop-down panel", () => {
    const store = mockStore(dataDump);
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <UserIcon />
        </Router>
      </Provider>
    );

    const userIconPanel = ".user-icon__panel";
    const userIconPanelToggle = ".p-icon--user";

    expect(wrapper.find(userIconPanel).hasClass("is-visible")).toEqual(false);
    wrapper.find(userIconPanelToggle).simulate("click");
    expect(wrapper.find(userIconPanel).hasClass("is-visible")).toEqual(true);
    wrapper.find(userIconPanelToggle).simulate("click");
    expect(wrapper.find(userIconPanel).hasClass("is-visible")).toEqual(false);
  });
});
