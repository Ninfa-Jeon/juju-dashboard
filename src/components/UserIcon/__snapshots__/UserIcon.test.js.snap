// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Icon renders without crashing and matches snapshot 1`] = `
<div
  className="user-icon"
>
  <i
    className="p-icon--user"
    onClick={[Function]}
  >
    Account icon
  </i>
  <div
    className="user-icon__panel p-card--highlighted"
  >
    <span
      className="user-icon__panel-arrow"
    />
    <ul
      className="p-list"
    >
      <li
        className="p-list__item"
      >
        <a
          href="#_"
        >
          Profile
        </a>
      </li>
      <li
        className="p-list__item"
      >
        <a
          href="#_"
        >
          Help
        </a>
      </li>
      <li
        className="p-list__item"
      >
        <Link
          onClick={[Function]}
          to="/"
        >
          <LinkAnchor
            href="/"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Log out
            </a>
          </LinkAnchor>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
